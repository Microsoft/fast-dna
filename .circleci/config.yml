# Validate this file using
# $ circleci config validate -c .circleci/config.yml
# Reference url: https://circleci.com/blog/circleci-hacks-validate-circleci-config-on-every-commit-with-a-git-hook/
version: 2
defaults: &defaults
  docker:
      - image: circleci/node:latest
  working_directory: ~/fast-dna

jobs:
  build:
    <<: *defaults
    steps:
      - checkout

      # Load installed dependencies from cache if possible, bundle install then save cache
      # Multiple caches are used to increase the chance of a cache hit
      - restore_cache:
          keys:
          - dependency-v1-{{ .Branch }}--{{ checksum "package-lock.json" }}          
          - dependency-v1-{{ .Branch }}--
          - dependency-v1-
        
      - run:
          name: Install the latest NPM version and patches
          command: |
            sudo npm i npm@latest -g
            npm update rc --depth 4
      - run:
          name: Install the latest Lerna version
          command: |
            sudo npm i lerna@^3.0.6 -g
      - run:
          name: Install all dependency packages
          command: |
            npm i
      - run:
          name: Execute Lerna bootstrap
          command: |
            lerna bootstrap
      
      # Save dependencies into cache
      - save_cache:
          paths:
            - ~/fast-dna/node_modules
            - ~/fast-dna/packages/**/node_modules
            - /usr/local/lib/node_modules
          key: dependency-v1-{{ .Branch }}--{{ checksum "package-lock.json" }}
      
 
  test:
    <<: *defaults
    steps:
      - run: 
          name: Unit Test all packages
          command: |
            lerna run test --stream
      - run:
          name: Setup and execute Code Climate reporter
          command: |
            if [ ! -f ./coverage/cc-test-reporter ]; then
              mkdir -p coverage/
              curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./coverage/cc-test-reporter
              chmod +x ./coverage/cc-test-reporter
            fi

            # Code Coverage
            # Notify Code Climate that a build is about to start
            ./coverage/cc-test-reporter before-build

            for f in packages/*; do
              if [ -d "$f" ]; then
                echo $f
                set +e
                if [ -d "$f/coverage" ]; then
                  ./coverage/cc-test-reporter format-coverage -t lcov -o coverage/coverage.${f//\//-}.json $f/coverage/lcov.info
                fi
              fi
            done;

            ./coverage/cc-test-reporter sum-coverage -o coverage/coverage.total.json coverage/coverage.*.json
            ./coverage/cc-test-reporter upload-coverage -i coverage/coverage.total.json
      - save_cache:
          key: ci-cache-{{ checksum "package.json" }}
          paths:
            - ./node_modules
            - ./coverage
            - ./coverage/cc-test-reporter


  # regress:
  #  docker:
  #    - image: circleci/node:latest
  #  working_directory: ~/fast-dna


workflows:
  version: 2
  build_and_test:
    jobs:
      - build
      - test:
          requires:
            - build
      
  # schedule-nightly-workflow:
  #   triggers:
  #     - schedule:
  #         cron: "0 0 * * *"
  #         filters:
  #           branches:
  #             only:
  #               - master
  #   jobs:
  #     - build
  #     - test:
  #         requires:
  #           - build
   #   - regress:
   #       requires:
   #         - build
   #         - test

            # Steps that need further wired out
              # regress
              # deploy
              # publish


# - run:
#   name: Install Browser Stack and Set Up Tunnel
#   background: true
#   command: |
#     # Download the browserstack binary file to create a tunnel
#     - wget "https://www.browserstack.com/browserstack-local/BrowserStackLocal-linux-x64.zip"
#     # Unzip the browserstack binary file
#     - unzip BrowserStackLocal-linux-x64.zip
#     # Run the file with your access key
#     - ./BrowserStackLocal $BROWSERSTACK KEY:
#         background: true

          # // TODO #428: Reconfigure to support workflows to improve performmance
#   "test-coverage":
#     environment:
#       CC_TEST_REPORTER_ID=08a773cb4ea5811add5a45e12873e5cd2634c005568705cc37abfd5217617a32
#     docker:
#       - image: circleci/node:latest
#     working_directory: ~/fast-dna
#     steps:
#       - checkout
#       - run:
#           name: Setup Code Climate test-reporter
#           command: |
#             mkdir -p coverage/
#             curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./coverage/cc-test-reporter
#             chmod +x ./coverage/cc-test-reporter
#       - run:
#           name: Run tests
#           command: |
#             ./coverage/cc-test-reporter before-build
#             for f in packages/*; do
#               if [ -d $f ]; then
#                  echo $f
#                 ./coverage/cc-test-reporter format-coverage -t lcov -o coverage/coverage.${f//\//-}.json $f/coverage/lcov.info
#               fi
#             done;
#             ./coverage/cc-test-reporter sum-coverage -o coverage/coverage.total.json coverage/coverage.*.json
#             ./coverage/cc-test-reporter upload-coverage -i coverage/coverage.total.json
#             ./coverage/cc-test-reporter after-build --coverage-input-type lcov --exit-code $?
# workflows:
#   version: 2
#   build-and-test:
#     jobs:
#       - build
#       - "test-coverage":
#           requires:
#             - build
