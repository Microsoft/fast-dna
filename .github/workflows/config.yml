name: Azure DevOps Actions

on:
  push:
    branches:
      - master

jobs:
  build_azure_pipelines:

    runs-on: ubuntu-latest
    
    steps:
    
    - name: Setup Node 10.x
      uses: actions/setup-node@v1
      with:
        node-version: '10.x'

    - name: 'Trigger an Azure Pipeline'
      uses: azure/pipelines@v1
      with:
        azure-devops-project-url: https://dev.azure.com/edgewebui/fast-dna
        azure-pipeline-name: fast.web.build
        azure-devops-token: ${{ secrets.AZURE_DEVOPS_TOKEN }}