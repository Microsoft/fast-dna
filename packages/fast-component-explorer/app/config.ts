import { MenuItem } from "@microsoft/fast-tooling-react";
import {
    actionToggleSchema,
    actionTriggerSchema,
    autoSuggestSchema,
    autoSuggestOptionSchema,
    badgeSchema,
    breadcrumbSchema,
    buttonSchema,
    callToActionSchema,
    captionSchema,
    cardSchema,
    carouselSchema,
    checkboxSchema,
    contextMenuSchema,
    contextMenuItemSchema,
    dialogSchema,
    dividerSchema,
    flipperSchema,
    headingSchema,
    hypertextSchema,
    imageSchema,
    labelSchema,
    metatextSchema,
    numberFieldSchema,
    paragraphSchema,
    pivotSchema,
    progressSchema,
    radioSchema,
    selectSchema,
    selectOptionSchema,
    sliderSchema,
    sliderLabelSchema,
    subheadingSchema,
    textActionSchema,
    textAreaSchema,
    textFieldSchema,
    toggleSchema,
    typographySchema,
} from "@microsoft/fast-components-react-msft";
import { createBrowserHistory } from "history";

const schemas: any[] = [
    actionToggleSchema,
    actionTriggerSchema,
    autoSuggestSchema,
    autoSuggestOptionSchema,
    badgeSchema,
    breadcrumbSchema,
    buttonSchema,
    callToActionSchema,
    captionSchema,
    cardSchema,
    carouselSchema,
    checkboxSchema,
    contextMenuSchema,
    contextMenuItemSchema,
    dialogSchema,
    dividerSchema,
    flipperSchema,
    headingSchema,
    hypertextSchema,
    imageSchema,
    labelSchema,
    metatextSchema,
    numberFieldSchema,
    paragraphSchema,
    pivotSchema,
    progressSchema,
    radioSchema,
    selectSchema,
    selectOptionSchema,
    sliderSchema,
    sliderLabelSchema,
    subheadingSchema,
    textActionSchema,
    textAreaSchema,
    textFieldSchema,
    toggleSchema,
    typographySchema,
];

function getRouteFromSchemaId(schemaId: string): string {
    const matchedRegex: RegExpMatchArray | null = schemaId.match(/\/(?:.(?!\/))+$/);
    return Array.isArray(matchedRegex) ? `/components${matchedRegex[0]}` : "";
}

function generateMenu(schemas: any[]): MenuItem[] {
    return [
        {
            displayName: "Components",
            items: schemas.map(
                (schema: any): MenuItem => {
                    return {
                        displayName: schema.title,
                        location: getRouteFromSchemaId(schema.id),
                    };
                }
            ),
        },
    ];
}

const history = createBrowserHistory();
const menu: MenuItem[] = generateMenu(schemas);

export { history, menu };
