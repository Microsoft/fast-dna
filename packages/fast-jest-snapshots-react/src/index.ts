import * as React from "react";
import * as renderer from "react-test-renderer";

/**
 * An interface describing component example objects used for snapshots and component testing.
 */
export interface ISnapshotTestSuite<T> {
    /**
     * The name of the component
     */
    name: string;

    /**
     * The component constructor
     */
    component: React.ComponentClass<T>;

    /**
     * Prop instances to pass to a component
     */
    data: T[];
}

/**
 * Executes a single snapshot test given a component, component props, and a test title
 */
export function renderSnapshot<T>(props: T, component: React.ComponentClass<T>, title: string): void {
    test(title, (): void => {
        const renderedComponent = renderer.create(React.createElement(component, props));
        const componentJson = renderedComponent.toJSON();

        expect(componentJson).toMatchSnapshot();
    });
}

/**
 * Generate a set of snapshot tests given a snapshot suite
 */
export function generateSnapshots<T>(examples: ISnapshotTestSuite<T>): void {
    const data: T[] = examples.data;
    const component: React.ComponentClass<T> = examples.component;

    if (Array.isArray(data)) {
        data.forEach((example: T, index: number): void => {
            renderSnapshot(example, component, `${examples.name}: ${index}`);
        });
    }
}
