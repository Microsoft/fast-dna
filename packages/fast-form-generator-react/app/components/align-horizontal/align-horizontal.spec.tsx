import * as React from "react";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { ISnapshotTestSuite } from "@microsoft/fast-jest-snapshots-react";
import { getExample } from "@microsoft/fast-permutator";
import Form, { IFormProps } from "../../../src/form";
import config from "./align-horizontal.config";
import * as alignHorizontalSchema from "./align-horizontal.schema.json";

const name: string = "align-horizontal";

const exampleData: IFormProps = {
    schema: alignHorizontalSchema,
    data: getExample(alignHorizontalSchema),
    onChange: (data: any): void => {},
    componentMappingToPropertyNames: config
}

const examples = {
    name,
    component: Form,
    data: [
        exampleData
    ]
};

describe(name, () => {
    generateSnapshots(examples);
});
