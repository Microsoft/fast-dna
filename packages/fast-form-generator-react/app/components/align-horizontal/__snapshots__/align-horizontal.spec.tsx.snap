// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align-horizontal align-horizontal: 0 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Component with align horizontal
    </h2>
    <div>
      <div>
        <div>
          <div>
            <div
              className="formItemAlignHorizontal-0-1-6"
            >
              <label
                className="formItemAlignHorizontal_label-0-1-7"
                htmlFor="alignHorizontal"
              >
                Align horizontal
              </label>
              <div>
                <span>
                  <input
                    aria-label="left align"
                    checked={true}
                    className="formItemAlignHorizontal_input__left-0-1-8"
                    id="alignHorizontal1"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="left"
                  />
                </span>
                <span>
                  <input
                    aria-label="center align"
                    checked={false}
                    className="formItemAlignHorizontal_input__center-0-1-9"
                    id="alignHorizontal2"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="center"
                  />
                </span>
                <span>
                  <input
                    aria-label="right align"
                    checked={false}
                    className="formItemAlignHorizontal_input__right-0-1-10"
                    id="alignHorizontal3"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="right"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
